jax/api.py

```diff
@@ -342,12 +392,11 @@ def abstractify(x):
 
   @wraps(fun)
   def computation_maker(*args, **kwargs):
-    wrapped = lu.wrap_init(fun)
-    if static_argnums:
-      dyn_argnums = [i for i in range(len(args)) if i not in static_argnums]
-      wrapped, _ = argnums_partial(wrapped, dyn_argnums, args)
+    f = lu.wrap_init(fun)
+    f, _ = apply_argnums(f, static_argnums, args)
+    f, _ = apply_argnames(f, static_argnames, kwargs)
```

Comment:
I'm a bit confused at this point... my understanding from above was that argnums and argnames now duplicate each other. Do we need to apply argnames if they've already been applied via their argnum determined from the signature?

Reply:
We need to handle both here, because we don't want to inspect function signatures each time a `jit` decorated function is called (`inspect` is slow). Thus `argnums` is applied to all parameters passed by position (`*args`), and ` argnames`  is applied to all parameters based by name (`**kwargs`).

Reply:
Ah, I see that now, thanks.